SurveyApp/Modules/Login/LoginViewController.swift

```diff
@@ -44,10 +49,26 @@ final class LoginViewController: UIViewController {
 
 extension LoginViewController: LoginViewInput {
 
+    var emailInputString: String? {
+        get { emailField.text }
+        set { emailField.text = newValue }
+    }
+
+    var passwordInputString: String? {
+        get { passwordField.text }
+        set { passwordField.text = newValue }
+    }
+    
     func configure() {
         setUpLayout()
         setUpViews()
     }
+    
+    func showError(message: String) {
```

Comment:
You may want to create a re-use solution for showing an error message, cause it will be used a lot in other modules
And repeat these codes are not the best option 😁 

My favourite solution that creates a protocol and extension is [where self](https://docs.swift.org/swift-book/LanguageGuide/Generics.html#ID553) is a view controller, so that you can add this alert error feature when you need

Reply:
Thank you. I'll make that change.